src/rtos/rtos_freertos_stackings.c

```diff
@@ -917,23 +923,31 @@ const struct rtos_register_stacking *rtos_freertos_riscv_pick_stacking_info(stru
 }
 
 // Chip-specific data for calculating Thread Local Storage (tls) address
-
-static const struct freertos_tls_info s_xtensa_tls = {
-	XT_REG_IDX_THREADPTR,	/* tls_reg */
-	16,			/* tls_align */
-};
+static struct freertos_tls_info s_xtensa_tls;
 
 static const struct freertos_tls_info s_riscv_tls = {
 	GDB_REGNO_TP,	/* tls_reg */
-	0,		/* tls_align */
+	0,				/* tls_align */
 };
 
 const struct freertos_tls_info *rtos_freertos_get_tls_info(struct target *target)
 {
 	if (strncmp(target_get_gdb_arch(target), "riscv", 5) == 0) {
 		return &s_riscv_tls;
 	} else if (strncmp(target_get_gdb_arch(target), "xtensa", 6) == 0) {
-		return &s_xtensa_tls;
+		struct reg **reg_list;
+		int num_regs;
+		int retval = target_get_gdb_reg_list(target, &reg_list, &num_regs, REG_CLASS_GENERAL);
+		if (retval != ERROR_OK)
+			return NULL;
+
+		for (int j = 0; j < num_regs; ++j) {
+			if (!strcmp(reg_list[j]->name, "threadptr")) {
+				s_xtensa_tls.tls_reg = j;
```

Comment:
@ianstcdns This is the example of getting reg_number dynamically. Please check just to confirm.

Reply:
looks reasonable, although you might consider strcasecmp to be somewhat more flexible.

Reply:
strcasecmp OK